### DISCUSSION TITLE
Collection primary key NanoId type

### DISCUSSION THREAD

**Discussion by tofran** [directus/directus#13662] [Feature Requests]
As of right now when creating a collection the primary key can be `Generated UUID`.
<img width="590" alt="image" src="https://user-images.githubusercontent.com/5692603/171381699-8d88adfd-6212-4cea-be9f-445c06b11f5e.png">

This works quite well, but as [NanoId](https://www.npmjs.com/package/nanoid) is getting more and more popular, would be nice if Directus offered an out of the box way of using it. I will not go into the details of why NanoId is a great alternative to UUID, you can read about that in [their readme.](https://github.com/ai/nanoid#comparison-with-uuid)

This addition should be quite smple. Fun fact, directus already imports nanoid (for token generation), so it would not add a new dependency:

https://github.com/directus/directus/blob/f7e70ede0cfaebbe5fbc9145b62bb2d09d95339b/package.json#L50

Does anyone think this is a good addition? Would it be feasible to implement this? Would this be a good for a first pull request?

**Comment by rijkvanzanten** [comment]
If we want to go that route, I would like to offer way more configuration for the primary key as well. In nanoid, you can choose the character set used, and length of the primary key for example. I'm also not sure how common it is to use nanoid in a sql database environment. UUIDs are often used now, as that's a dedicated type in more and more SQL databases (which comes with it's own improvements), whereas nanoid would effectively be a "regular" string of a fixed length 🤔

**Reply by tofran** [reply]
Good points

> dedicated type in more and more SQL databases (which comes with it's own improvements)

This is actually the reason why I haven't used nanoid in production. Still, would be cool to see it in Directus.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
